cmd/mobynit/main.go

```diff
@@ -65,9 +66,72 @@ func mountContainer(containerID, graphDriver string) string {
 	return newRootPath
 }
 
+func mountBalenaContainerLocal(layer_root, mount_dest string) string {
+
+	rawGraphDriver, err := ioutil.ReadFile(filepath.Join(layer_root,"/current/boot/storage-driver"))
+	if err != nil {
+		log.Fatal("could not get storage driver:", err)
+	}
+	graphDriver := strings.TrimSpace(string(rawGraphDriver))
+
+	current, err := os.Readlink(filepath.Join(layer_root,"/current"))
+	if err != nil {
+		log.Fatal("could not get container ID:", err)
+	}
+	containerID := filepath.Base(current)
+
+	if err := os.MkdirAll(mount_dest, os.ModePerm); err != nil {
+		log.Fatal("creating %s failed:", mount_dest, err)
+	}
+
+	ls, err := layer.NewStoreFromOptions(layer.StoreOptions{
+		Root:                      filepath.Join(layer_root,"/balena"),
+		MetadataStorePathTemplate: filepath.Join(layer_root, "balena", "image", "%s", "layerdb"),
+		IDMappings:                &idtools.IDMappings{},
+		GraphDriver:               graphDriver,
+		OS:                        "linux",
+	})
+	if err != nil {
+		log.Fatal("error loading layer store:", err)
+	}
+
+	rwlayer, err := ls.GetRWLayer(containerID)
+	if err != nil {
+		log.Fatal("error getting container layer:", err)
+	}
+
+	rwlayer.Unmount()
+	newRoot, err := rwlayer.Mount(mount_dest)
+	if err != nil {
+		log.Fatal("error mounting container fs:", err)
+	}
+	newRootPath := newRoot.Path()
+
+	if err := unix.Mount("", newRootPath, "", unix.MS_REMOUNT, ""); err != nil {
+		log.Fatal("error remounting container as read/write:", err)
+	}
+	unix.Mount("", newRootPath, "", unix.MS_REMOUNT|unix.MS_RDONLY, "")
```

Comment:
Here you first remount rw and immediately after you mount it read-only again. It's a big no-op

Reply:
Indeed. Removed. Inherited from the mobynit mountContainer function